ocaml/xapi/xapi_xenops.ml

```diff
@@ -2016,6 +2016,8 @@ let on_xapi_restart ~__context =
 		Client.VM.remove dbg id
 	) xenopsd_vms_not_in_xapi;
 
+	List.iter (fun ((id, _), _) -> add_caches id) xenopsd_vms_in_xapi;
```

Comment:
I think this should be moved down into the code that's already going through `xenopsd_vms_in_xapi` in the branch:

```| `Running | `Paused ->...```